### PR TITLE
HL7 Support

### PR DESCRIPTION
**Pull request by RickHawesUSDS**
This PR does adds support to send HL7 messages to our partners. 

## Dev Notes
Had to define new element types and enhance existing types. I've added the LIVD for HHS guidance conformance and FIPS tables for County validation. 

The NIST validator was used to ensure conformance. 

## Changes
- TABLE element type
- Support for OBX segment including the AOE OBX type
- Support for the NTE segment
- FakeReport updated to draw from the new table
- Commented out the unused elements from the covid-19.schema
- AZ-HL7 schema updated

## Checklist
- [x] Tested locally?
- [x] Added new dependencies?
- [ ] Updated the docs?
- [x] Added a test?
- [ ] Did you check for sensitive data, and remove any?
- [ ] Are additional approvals needed for this change?
- [ ] Are there potential vulnerabilities or licensing issues with any new dependencies introduced?

## Fixes 

- #144 
- #10 

## To Be Done
- Reading HL7

### CODE REVIEW

File: prime-router/metadata/schemas/covid-19.schema [hunk 1]
```diff
@@ -635,218 +705,203 @@ elements:
...
-    type: TEXT
-    natFlatFileField: Test_result_number2
-    hhsGuidanceField:
-    hl7Field: #OBX-5.4
+    hl7Field: OBX-22
 
   - name: test_result_status
     type: CODE
+    default: F
     valueSet: hl70123
```
Comment (RickHawesUSDS) [inline prime-router/metadata/schemas/covid-19.schema:800]: F means Final

File: prime-router/src/main/kotlin/Element.kt [hunk 2]
```diff
@@ -214,42 +239,45 @@ data class Element(
...
                 } catch (e: DateTimeParseException) {
                     null
+                    // Try a HL7 pattern
                 } ?: try {
-                    val formatter = DateTimeFormatter.ofPattern(field?.format ?: datetimePattern, Locale.ENGLISH)
+                    val formatter = DateTimeFormatter.ofPattern(format ?: datetimePattern, Locale.ENGLISH)
                     OffsetDateTime.parse(formattedValue, formatter)
                 } catch (e: DateTimeParseException) {
                     null
+                    // Finally, accept a local date pattern and assume it is in the central timezone (small amount of error)
```
Comment (jimduff-usds) [inline prime-router/src/main/kotlin/Element.kt:261]: Yes, this is the slot where the issue I found yesterday lies.

File: prime-router/src/main/kotlin/Hl7Converter.kt [hunk 3]
```diff
@@ -52,23 +52,33 @@ object Hl7Converter {
     }
```
Comment (jimduff-usds) [inline prime-router/src/main/kotlin/Hl7Converter.kt:52]: I need to spend some time playing around with HL7 conversions.

File: prime-router/src/main/kotlin/Mappers.kt [hunk 4]
```diff
@@ -21,43 +21,54 @@ interface Mapper {
 
     /**
      *
-     * The elements that the mapper requests for the apply function
```
Comment (jimduff-usds) [inline prime-router/src/main/kotlin/Mappers.kt:24]: An example would be helpful here

**Review by jimduff-usds — APPROVED** [review]
I did pull, compile and run old csv routings, works fine.
Have not tested HL7 creation at all.   Should we put some test HL7 receiver Orgs in organizations.yml?

File: prime-router/src/main/kotlin/Mappers.kt (hunk 4 above)
Comment (RickHawesUSDS) [reply prime-router/src/main/kotlin/Mappers.kt:24]: Added an example.

**Pull request merged**